ModMenu does not work on snapshot 22w45a

```
Description: Initializing game

java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_757 failed
	at net.minecraft.class_4668.<clinit>(class_4668.java:125)
	at net.minecraft.class_380.<init>(class_380.java:29)
	at net.minecraft.class_377.method_2012(class_377.java:163)
	at net.minecraft.class_7191.bake(class_7191.java:53)
	at net.minecraft.class_377.method_2004(class_377.java:53)
	at net.minecraft.class_378.method_27540(class_378.java:45)
	at net.minecraft.class_156.method_654(class_156.java:440)
	at net.minecraft.class_378.<init>(class_378.java:45)
	at net.minecraft.class_310.<init>(class_310.java:536)
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_757 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:419)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:145)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 14 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:414)
	... 18 more
Caused by: org.spongepowered.asm.mixin.injection.throwables.InjectionError: Critical injection failure: Callback method onBeforeRenderScreen(FJZLorg/spongepowered/asm/mixin/injection/callback/CallbackInfo;IILnet/minecraft/class_4587;)V in fabric-screen-api-v1.mixins.json:GameRendererMixin from mod fabric-screen-api-v1 failed injection check, (0/1) succeeded. Scanned 1 target(s). Using refmap client-fabric-screen-api-v1-refmap.json
	at org.spongepowered.asm.mixin.injection.struct.InjectionInfo.postInject(InjectionInfo.java:468)
	at org.spongepowered.asm.mixin.transformer.MixinTargetContext.applyInjections(MixinTargetContext.java:1385)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyInjections(MixinApplicatorStandard.java:1052)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.applyMixin(MixinApplicatorStandard.java:400)
	at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:325)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 21 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_4668.<clinit>(class_4668.java:125)
	at net.minecraft.class_380.<init>(class_380.java:29)
	at net.minecraft.class_377.method_2012(class_377.java:163)
	at net.minecraft.class_7191.bake(class_7191.java:53)
	at net.minecraft.class_377.method_2004(class_377.java:53)
	at net.minecraft.class_378.method_27540(class_378.java:45)
	at net.minecraft.class_156.method_654(class_156.java:440)
	at net.minecraft.class_378.<init>(class_378.java:45)
	at net.minecraft.class_310.<init>(class_310.java:536)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.main.Main.method_44604(Main.java:205)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 22w45a
	Minecraft Version ID: 22w45a
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 379511808 bytes (361 MiB) / 1140850688 bytes (1088 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 4
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i3-8100 CPU @ 3.60GHz
	Identifier: Intel64 Family 6 Model 158 Stepping 11
	Microarchitecture: Coffee Lake
	Frequency (GHz): 3.60
	Number of physical packages: 1
	Number of physical CPUs: 4
	Number of logical CPUs: 4
	Graphics card #0 name: NVIDIA GeForce GT 710
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 1024.00
	Graphics card #0 deviceId: 0x128b
	Graphics card #0 versionInfo: DriverVersion=27.21.14.6109
	Memory slot #0 capacity (MB): 4096.00
	Memory slot #0 clockSpeed (GHz): 2.40
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 4096.00
	Memory slot #1 clockSpeed (GHz): 2.40
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 13259.00
	Virtual memory used (MB): 9082.53
	Swap memory total (MB): 5120.00
	Swap memory used (MB): 263.84
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Launched Version: fabric-loader-0.14.10-22w45a
	Backend library: LWJGL version 3.3.1 SNAPSHOT
	Backend API: GeForce GT 710/PCIe/SSE2 GL version 3.2.0 NVIDIA 461.09, NVIDIA Corporation
	Window size: <not initialized>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: 4x Intel(R) Core(TM) i3-8100 CPU @ 3.60GHz
```

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,10 +3,10 @@ org.gradle.jvmargs=-Xmx1G
 maven_group=com.terraformersmc
 archive_name=modmenu
 
-minecraft_version=22w43a
-yarn_mappings=22w43a+build.7
+minecraft_version=22w45a
+yarn_mappings=22w45a+build.9
 loader_version=0.14.10
-fabric_version=0.65.1+1.19.3
+fabric_version=0.66.1+1.19.3
 quilt_loader_version=0.17.1
 
 # Project Metadata

--- a/src/main/java/com/terraformersmc/modmenu/gui/ModMenuOptionsScreen.java
+++ b/src/main/java/com/terraformersmc/modmenu/gui/ModMenuOptionsScreen.java
@@ -9,11 +9,7 @@ import net.minecraft.client.gui.widget.ButtonListWidget;
 import net.minecraft.client.gui.widget.ButtonWidget;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.screen.ScreenTexts;
-import net.minecraft.text.OrderedText;
 import net.minecraft.text.Text;
-
-import java.util.List;
-import java.util.function.Supplier;
 
 public class ModMenuOptionsScreen extends GameOptionsScreen {
 
@@ -44,11 +40,6 @@ public class ModMenuOptionsScreen extends GameOptionsScreen {
 		this.list.render(matrices, mouseX, mouseY, delta);
 		drawCenteredText(matrices, this.textRenderer, this.title, this.width / 2, 5, 0xffffff);
 		super.render(matrices, mouseX, mouseY, delta);
-		List<OrderedText> list = getHoveredButtonTooltip(this.list, mouseX, mouseY);
-		if (list != null) {
-			this.renderOrderedTooltip(matrices, list, mouseX, mouseY);
-		}
-
 	}
 
 	public void removed() {

--- a/src/main/java/com/terraformersmc/modmenu/gui/ModsScreen.java
+++ b/src/main/java/com/terraformersmc/modmenu/gui/ModsScreen.java
@@ -14,6 +14,7 @@ import com.terraformersmc.modmenu.util.TranslationUtil;
 import com.terraformersmc.modmenu.util.mod.Mod;
 import com.terraformersmc.modmenu.util.mod.ModBadgeRenderer;
 import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.class_7919;
 import net.minecraft.client.gui.DrawableHelper;
 import net.minecraft.client.gui.screen.ConfirmLinkScreen;
 import net.minecraft.client.gui.screen.ConfirmScreen;
@@ -56,7 +57,6 @@ public class ModsScreen extends Screen {
 	private DescriptionListWidget descriptionListWidget;
 	private final Screen previousScreen;
 	private ModListWidget modList;
-	private Text tooltip;
 	private ModListEntry selected;
 	private ModBadgeRenderer modBadgeRenderer;
 	private double scrollPercent = 0;
@@ -134,14 +134,6 @@ public class ModsScreen extends Screen {
 			} else {
 				button.active = false;
 			}
-		},
-				CONFIGURE, (buttonWidget, matrices, mouseX, mouseY) -> {
-			ModMenuTexturedButtonWidget button = (ModMenuTexturedButtonWidget) buttonWidget;
-			if (button.isJustHovered()) {
-				this.renderTooltip(matrices, CONFIGURE, mouseX, mouseY);
-			} else if (button.isFocusedButNotHovered()) {
-				this.renderTooltip(matrices, CONFIGURE, button.getX(), button.getY());
-			}
 		}) {
 			@Override
 			public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
@@ -162,6 +154,7 @@ public class ModsScreen extends Screen {
 				super.renderButton(matrices, mouseX, mouseY, delta);
 			}
 		};
+		configureButton.method_47400(class_7919.method_47407(CONFIGURE));
 		int urlButtonWidths = paneWidth / 2 - 2;
 		int cappedButtonWidth = Math.min(urlButtonWidths, 200);
 		ButtonWidget websiteButton = new ButtonWidget(rightPaneX + (urlButtonWidths / 2) - (cappedButtonWidth / 2), paneY + 36, Math.min(urlButtonWidths, 200), 20,
@@ -173,7 +166,7 @@ public class ModsScreen extends Screen {
 				}
 				this.client.setScreen(this);
 			}, mod.getWebsite(), false));
-		}, ButtonWidget.EMPTY_TOOLTIP, Supplier::get) {
+		}, Supplier::get) {
 			@Override
 			public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
 				visible = selected != null;
@@ -190,7 +183,7 @@ public class ModsScreen extends Screen {
 				}
 				this.client.setScreen(this);
 			}, mod.getIssueTracker(), false));
-		}, ButtonWidget.EMPTY_TOOLTIP, Supplier::get) {
+		}, Supplier::get) {
 			@Override
 			public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
 				visible = selected != null;
@@ -199,14 +192,9 @@ public class ModsScreen extends Screen {
 			}
 		};
 		this.addSelectableChild(this.searchBox);
-		this.addDrawableChild(new ModMenuTexturedButtonWidget(paneWidth / 2 + searchBoxWidth / 2 - 20 / 2 + 2, 22, 20, 20, 0, 0, FILTERS_BUTTON_LOCATION, 32, 64, button -> filterOptionsShown = !filterOptionsShown, TOGGLE_FILTER_OPTIONS, (buttonWidget, matrices, mouseX, mouseY) -> {
-			ModMenuTexturedButtonWidget button = (ModMenuTexturedButtonWidget) buttonWidget;
-			if (button.isJustHovered()) {
-				this.renderTooltip(matrices, TOGGLE_FILTER_OPTIONS, mouseX, mouseY);
-			} else if (button.isFocusedButNotHovered()) {
-				this.renderTooltip(matrices, TOGGLE_FILTER_OPTIONS, button.getX(), button.getY());
-			}
-		}));
+		ButtonWidget filtersButton = new ModMenuTexturedButtonWidget(paneWidth / 2 + searchBoxWidth / 2 - 20 / 2 + 2, 22, 20, 20, 0, 0, FILTERS_BUTTON_LOCATION, 32, 64, button -> filterOptionsShown = !filterOptionsShown, TOGGLE_FILTER_OPTIONS);
+		filtersButton.method_47400(class_7919.method_47407(TOGGLE_FILTER_OPTIONS));
+		this.addDrawableChild(filtersButton);
 		Text showLibrariesText = ModMenuConfig.SHOW_LIBRARIES.getButtonText();
 		Text sortingText = ModMenuConfig.SORTING.getButtonText();
 		int showLibrariesWidth = textRenderer.getWidth(showLibrariesText) + 20;
@@ -218,7 +206,7 @@ public class ModsScreen extends Screen {
 			ModMenuConfig.SORTING.cycleValue();
 			ModMenuConfigManager.save();
 			modList.reloadFilters();
-		}, ButtonWidget.EMPTY_TOOLTIP, Supplier::get) {
+		}, Supplier::get) {
 			@Override
 			public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
 				matrices.translate(0, 0, 1);
@@ -231,7 +219,7 @@ public class ModsScreen extends Screen {
 			ModMenuConfig.SHOW_LIBRARIES.toggleValue();
 			ModMenuConfigManager.save();
 			modList.reloadFilters();
-		}, ButtonWidget.EMPTY_TOOLTIP, Supplier::get) {
+		}, Supplier::get) {
 			@Override
 			public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
 				matrices.translate(0, 0, 1);
@@ -251,14 +239,12 @@ public class ModsScreen extends Screen {
 				ButtonWidget.createBuilder(Text.translatable ("modmenu.modsFolder"), button -> Util.getOperatingSystem().open(new File(FabricLoader.getInstance().getGameDir().toFile(), "mods")))
 						.setPosition(this.width / 2 - 154, this.height - 28)
 						.setSize(150, 20)
-						.setTooltipSupplier(ButtonWidget.EMPTY_TOOLTIP)
 						.setNarrationSupplier(Supplier::get)
 						.build());
 		this.addDrawableChild(
 				ButtonWidget.createBuilder(ScreenTexts.DONE, button -> client.setScreen(previousScreen))
 						.setPosition(this.width / 2 + 4, this.height - 28)
 						.setSize(150, 20)
-						.setTooltipSupplier(ButtonWidget.EMPTY_TOOLTIP)
 						.setNarrationSupplier(Supplier::get)
 						.build());
 		this.setInitialFocus(this.searchBox);
@@ -279,7 +265,6 @@ public class ModsScreen extends Screen {
 	@Override
 	public void render(MatrixStack matrices, int mouseX, int mouseY, float delta) {
 		this.renderBackground(matrices);
-		this.tooltip = null;
 		ModListEntry selectedEntry = selected;
 		if (selectedEntry != null) {
 			this.descriptionListWidget.render(matrices, mouseX, mouseY, delta);
@@ -330,7 +315,7 @@ public class ModsScreen extends Screen {
 			}
 			textRenderer.draw(matrices, Language.getInstance().reorder(trimmedName), x + imageOffset, paneY + 1, 0xFFFFFF);
 			if (mouseX > x + imageOffset && mouseY > paneY + 1 && mouseY < paneY + 1 + textRenderer.fontHeight && mouseX < x + imageOffset + textRenderer.getWidth(trimmedName)) {
-				setTooltip(Text.translatable("modmenu.modIdToolTip", mod.getId()));
+				this.method_47415(Text.translatable("modmenu.modIdToolTip", mod.getId()));
 			}
 			if (init || modBadgeRenderer == null || modBadgeRenderer.getMod() != mod) {
 				modBadgeRenderer = new ModBadgeRenderer(x + imageOffset + this.client.textRenderer.getWidth(trimmedName) + 2, paneY, width - 28, selectedEntry.mod, this);
@@ -355,9 +340,6 @@ public class ModsScreen extends Screen {
 			}
 		}
 		super.render(matrices, mouseX, mouseY, delta);
-		if (this.tooltip != null) {
-			this.renderOrderedTooltip(matrices, textRenderer.wrapLines(this.tooltip, Integer.MAX_VALUE), mouseX, mouseY);
-		}
 	}
 
 	private Text computeModCountText(boolean includeLibs) {
@@ -412,10 +394,6 @@ public class ModsScreen extends Screen {
 	public void close() {
 		this.modList.close();
 		this.client.setScreen(this.previousScreen);
-	}
-
-	private void setTooltip(Text tooltip) {
-		this.tooltip = tooltip;
 	}
 
 	public ModListEntry getSelectedEntry() {

--- a/src/main/java/com/terraformersmc/modmenu/gui/widget/ModMenuButtonWidget.java
+++ b/src/main/java/com/terraformersmc/modmenu/gui/widget/ModMenuButtonWidget.java
@@ -8,6 +8,6 @@ import net.minecraft.text.Text;
 
 public class ModMenuButtonWidget extends ButtonWidget {
 	public ModMenuButtonWidget(int x, int y, int width, int height, Text text, Screen screen) {
-		super(x, y, width, height, text, button -> MinecraftClient.getInstance().setScreen(new ModsScreen(screen)), ButtonWidget.EMPTY_TOOLTIP, ButtonWidget.DEFAULT_NARRATION_SUPPLIER);
+		super(x, y, width, height, text, button -> MinecraftClient.getInstance().setScreen(new ModsScreen(screen)), ButtonWidget.DEFAULT_NARRATION_SUPPLIER);
 	}
 }

--- a/src/main/java/com/terraformersmc/modmenu/gui/widget/ModMenuTexturedButtonWidget.java
+++ b/src/main/java/com/terraformersmc/modmenu/gui/widget/ModMenuTexturedButtonWidget.java
@@ -24,14 +24,11 @@ public class ModMenuTexturedButtonWidget extends ButtonWidget {
 	}
 
 	public ModMenuTexturedButtonWidget(int x, int y, int width, int height, int u, int v, Identifier texture, int uWidth, int vHeight, PressAction onPress, Text message) {
-		this(x, y, width, height, u, v, texture, uWidth, vHeight, onPress, message, ButtonWidget.EMPTY_TOOLTIP);
+		this(x, y, width, height, u, v, texture, uWidth, vHeight, onPress, message, ButtonWidget.DEFAULT_NARRATION_SUPPLIER);
 	}
 
-	public ModMenuTexturedButtonWidget(int x, int y, int width, int height, int u, int v, Identifier texture, int uWidth, int vHeight, PressAction onPress, Text message, TooltipSupplier tooltipSupplier) {
-		this(x, y, width, height, u, v, texture, uWidth, vHeight, onPress, message, tooltipSupplier, ButtonWidget.DEFAULT_NARRATION_SUPPLIER);
-	}
-	public ModMenuTexturedButtonWidget(int x, int y, int width, int height, int u, int v, Identifier texture, int uWidth, int vHeight, PressAction onPress, Text message, TooltipSupplier tooltipSupplier, NarrationSupplier narationSupplier) {
-		super(x, y, width, height, message, onPress, tooltipSupplier, narationSupplier);
+	public ModMenuTexturedButtonWidget(int x, int y, int width, int height, int u, int v, Identifier texture, int uWidth, int vHeight, PressAction onPress, Text message, NarrationSupplier narationSupplier) {
+		super(x, y, width, height, message, onPress, narationSupplier);
 		this.uWidth = uWidth;
 		this.vHeight = vHeight;
 		this.u = u;
@@ -53,17 +50,5 @@ public class ModMenuTexturedButtonWidget extends ButtonWidget {
 
 		drawTexture(matrices, this.getX(), this.getY(), this.u, adjustedV, this.width, this.height, this.uWidth, this.vHeight);
 		RenderSystem.enableDepthTest();
-
-		if (this.isHovered()) {
-			this.renderTooltip(matrices, mouseX, mouseY);
-		}
-	}
-
-	public boolean isJustHovered() {
-		return hovered;
-	}
-
-	public boolean isFocusedButNotHovered() {
-		return !hovered && isFocused();
 	}
 }

